src/Symfony/Component/Messenger/Transport/Doctrine/DoctrineSender.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\Doctrine;
+
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Transport\Sender\SenderInterface;
+use Symfony\Component\Messenger\Transport\Serialization\Serializer;
+use Symfony\Component\Messenger\Transport\Serialization\SerializerInterface;
+
+/**
+ * @author Vincent Touzet <vincent.touzet@gmail.com>
+ */
+class DoctrineSender implements SenderInterface
+{
+    private $connection;
+    private $serializer;
+
+    public function __construct(Connection $connection, SerializerInterface $serializer = null)
+    {
+        $this->connection = $connection;
+        $this->serializer = $serializer ?? Serializer::create();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function send(Envelope $envelope): Envelope
+    {
+        $encodedMessage = $this->serializer->encode($envelope);
+
+        $this->connection->publish($encodedMessage['body'], $encodedMessage['headers'] ?? []);
```

Comment:
We should catch this and throw the `TransportException` like in the receiver.'

Also, we're missing `DelayStamp` support. I know, there were 1000 moving pieces, and you've done a great job quickly keeping up with them :). Check `AmqpSender`

Reply:
Hmm, very good point. It'll just need to be something we document. The default setting in Messenger uses "whole" seconds, anyways, so you'd need to customize it too use ms.

Reply:
Added a todo in https://github.com/symfony/symfony-docs/pull/10616 to document this